src/effects/audiounits/AudioUnitUtils.h

```diff
@@ -15,6 +15,23 @@
 #include <AudioUnit/AudioUnit.h>
 
 namespace AudioUnitUtils {
+   //! Type-erased function to get an AudioUnit property of fixed size
+   OSStatus GetFixedSizePropertyPtr(AudioUnit unit, AudioUnitPropertyID inID,
+      void *pProperty, UInt32 size, AudioUnitScope inScope,
+      AudioUnitElement inElement);
+
+   //! Get an AudioUnit property of deduced type and fixed size,
+   //! supplying most often used values as defaults for scope and element
+   template<typename T>
+   OSStatus GetFixedSizeProperty(AudioUnit unit, AudioUnitPropertyID inID,
```

Comment:
I like these helpers, and I was wondering whether instead of returning an OSStatus (which I understand is 0 for success) they could just return a bool - this because the if statements which use these helpers, at first sight display an "inverted logic" which causes fatigue for the reader of the code.

Reply:
It seems some inconsistent inverted logic is unavoidable so long as we call other functions in the AudioUnits sdk that return a 0 value for success.

In principle OSStatus could distinguish different kinds of errors with different values, but funny I can't find any enumeration of those values in the header files, just a typedef of OSStatus as SInt32.  🤷